pkg/apis/ceph.rook.io/v1/placement.go

```diff
@@ -130,7 +141,27 @@ func GetArbiterPlacement(p PlacementSpec) Placement {
 	return p[KeyMonArbiter]
 }
 
+func (p Placement) MergeOsd(with Placement) Placement {
+	var ret Placement
+	if with.PodAffinity != nil {
+		ret.PodAffinity = with.PodAffinity
+	}
+	if with.PodAntiAffinity != nil {
+		ret.PodAntiAffinity = with.PodAntiAffinity
+	}
+	if with.TopologySpreadConstraints != nil {
+		ret.TopologySpreadConstraints = with.TopologySpreadConstraints
+	}
+	if with.Tolerations != nil || p.Tolerations != nil {
+		ret.Tolerations = p.mergeTolerations(with.Tolerations)
+	}
+	if with.NodeAffinity != nil {
+		ret.NodeAffinity = with.NodeAffinity
```

Comment:
still, this two line of code is not perfect or have some issue. It is overriding node affinity when discovering nodes, but I don't think it will cause a problem.

Reply:
Why do we need a separate method for `MergeOsd()`? Seems like merging the tolerations should always be done in the `Merge()` method so it would apply to all daemons, then no need for this method.

Reply:
agreed, merging toleration in `merge()` method will be better, and also it will be applied to other pods like OSD, MGR, MON